src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala

```diff
@@ -459,6 +459,14 @@ trait ContextErrors {
       }
 
       //typedEta
+      def UnderscoreNullaryEtaMessage(tpe: Type) =
+        s"Only function types can be followed by _ but the current expression has type $tpe"
```

Comment:
I think the message is OK for new code being written, but since we're introducing a new warning that can trigger on existing code, we can be a bit more verbose?

Also, use "method" instead of "function"?

```suggestion
        s"Methods without a parameter list can no longer be converted to functions as `m _`, write a function literal `() => m` instead"
```

We'd need another version for by-name params.